File: python-martian-ddd/src/martian_ddd/domain/scents.py
from __future__ import annotations
from .types import Position, Orientation


class ScentRepository:
    def __init__(self) -> None:
        self._set: set[tuple[int, int, Orientation]] = set()

    def has(self, pos: Position, orientation: Orientation) -> bool:
        return (pos.x, pos.y, orientation) in self._set

    def add(self, pos: Position, orientation: Orientation) -> None:
        self._set.add((pos.x, pos.y, orientation))

File: node-martian-ddd/src/domain/scents.js
"use strict";

class ScentRepository {
  constructor() {
    this.set = new Set();
  }
  has(pos, o) {
    return this.set.has(`${pos.x},${pos.y},${o}`);
  }
  add(pos, o) {
    this.set.add(`${pos.x},${pos.y},${o}`);
  }
}

module.exports = { ScentRepository };
